portal/views/healthcheck.py

```diff
@@ -0,0 +1,138 @@
+from datetime import datetime, timedelta
+from flask import Blueprint, current_app
+import os
+import redis
+from redis import ConnectionError
+from sqlalchemy import text
+import subprocess
+
+from ..database import db
+
+
+healthcheck_blueprint = Blueprint('healthcheck', __name__)
+last_celery_beat_ping = None
+
+
+@healthcheck_blueprint.route('/celery_beat_ping')
+def celery_beat_ping():
+    """Periodically called by a celery beat task
+
+    Updates the last time we recieved a call to this API.
+    This allows us to monitor whether celery beat tasks are running
+    """
+    global last_celery_beat_ping
+    last_celery_beat_ping = datetime.now()
+    return 'PONG'
+
+
+def get_celery_beat_threshold():
+    """A timedelta representing max time we can go without a ping"""
+    # The interval celery beat pings the /celery_beat_ping API
+    ping_interval = current_app.config['CELERY_BEAT_PING_INTERVAL_SECONDS']
+
+    # The number of times we can miss a ping before we fail
+    missed_beats_before_fail = \
+        current_app.config['CELERY_BEAT_MISSED_PINGS_BEFORE_FAIL']
+
+    # The maximum amount of time we can go
+    # without a ping and still succeed
+    threshold = timedelta(
+        seconds=(ping_interval * missed_beats_before_fail)
+    )
+
+    return threshold
+
+
+##############################
+# Healthcheck functions below
+##############################
+
+def celery_available():
+    """Determines whether celery is available"""
+    # Ping celery. If we get a response
+    # then celery is available. Otherwise
+    # it is not available
+    FNULL = open(os.devnull, 'w')
+    code = subprocess.call([
```

Comment:
I have a feeling this won't work in production since Celery isn't required to run on the same machine. I'm not sure how else to check whether celery is running other than creating an API on the celery process that we ping from our APIs. But that method seems like overkill, and even unneccessary since we have the celery_beat_available() check that would fail is celery isn't running.

@pbugni @ivan-c do you guys have any thoughts on this? I'm planning on removing this check unless you think this will work or if there is a good alternate method available.

Reply:
I'd suggest you simply make a request to the celery test endpoint to confirm a simply task (addition) can be executed by celery:  `/celery-test&redirect-to-result=True`

Reply:
awesome! Will do.